pkg/pillar/types/locationconsts.go

```diff
@@ -180,3 +183,47 @@ var (
 	// it is not a constant so tests can override it.
 	TpmMeasurefsEventLog = PersistStatusDir + "/measurefs_tpm_event_log"
 )
+
+func ServerFileName() string {
+	return serverFileName
+}
+
+func Server() (string, error) {
+	bs, err := os.ReadFile(ServerFileName())
+	server := string(bs)
+	server = strings.TrimSpace(server)
+
+	return server, err
+}
+
+type errAndWarnLogger interface {
+	Error(args ...interface{})
+	Warnf(format string, args ...interface{})
+}
+
+func WaitServer(log errAndWarnLogger, stillRunning func()) string {
+	for {
+		s, err := Server()
+		if err != nil {
+			log.Error(err)
+		}
+
+		if len(s) != 0 {
+			return s
+		}
+
+		log.Warnf("Empty %s file - waiting for it",
+			ServerFileName())
+
+		stillRunning()
+		time.Sleep(10 * time.Second)
+	}
+}
+
+func WriteServer(newServer string) error {
+	if err := os.WriteFile(ServerFileName(), []byte(newServer), 0644); err != nil {
```

Comment:
Are you using this?

It can't work for those reasons:
1. /config is an overlay mount i.e., memory. Thus any changes are lost on reboot. (see eve config mount command)
2. To be robust against random power outages any changes to files on disk should use WriteRename, and the reader might need to look for a backup file in case the new file is empty.